crates/astria-conductor/src/sequencer/mod.rs

```diff
@@ -99,6 +103,18 @@ impl Reader {
 
     #[instrument(skip_all, err)]
     async fn initialize(&mut self) -> eyre::Result<executor::Handle<StateIsInit>> {
+        let actual_sequencer_chain_id =
+            get_sequencer_chain_id(self.sequencer_cometbft_client.clone())
+                .await
+                .wrap_err("failed to get chain ID from Sequencer")?
+                .to_string();
```

Comment:
Why allocate? It provides a way to get a str slice which you can check against: https://docs.rs/tendermint/latest/tendermint/chain/id/struct.Id.html#method.as_str